cancermuts/datasources.py

```diff
@@ -773,7 +759,14 @@ def _parse_db_files(self, gene_id, genome_assembly_version = 'GRCh38',
 
             df = self._load_db_files(filtered_targeted_database_file, filtered_screenmut_database_file)
 
-            df = df[ (df['IS_CANONICAL'] == 'y') ]
+        if transcript_accession:
+            self.log.info(f"Filtering by transcript_accession={transcript_accession}")
+            if 'TRANSCRIPT_ACCESSION' in df.columns:
+                df = df[df['TRANSCRIPT_ACCESSION'] == transcript_accession]
+                if df.empty:
+                    self.log.warning(f"No COSMIC rows for gene {gene_id} with TRANSCRIPT_ACCESSION={transcript_accession}; falling back to gene-level rows")
```

Comment:
can we change the wording from rows to mutations? So it's more clear for users

also I don't think we should fall-back, this is error prone; I think we should just continue and return an empty dataframe